common/config/loader.go

```diff
@@ -103,6 +105,21 @@ func Load(env string, configDir string, zone string, config interface{}) error {
 		}
 	}
 
+	// Inject self-dynamic-config-client if dynamicConfig section present.
+	if config.DynamicConfig != nil {
+		if config.DynamicConfigClient == nil {
```

Comment:
When is it okay for this not to be nil if dynamic config is specified here?

Reply:
I made it allowed to leave out dynamicConfigClient.. the only thing in there is pollInterval, if that's zero it'll just not do the refresh loop.

Reply:
Oh your question was the opposite. Then the answer is: there should be a way to specify nonzero pollInterval if you want to get updates. It's easiest to just use the existing setting.